adc/ad7779/ad7779-driver.c

```diff
@@ -80,11 +134,38 @@ static int dev_close(oid_t *oid, int flags)
 
 	return EOK;
 }
+#ifdef AD7779_SUPPORT_ASYNC_REQS
+static int dev_read(async_req_t *req, int *respond)
+{
+	*respond = 1;
 
-static int dev_read(void *data, size_t size)
+	if (ad7779_common.enabled != 1) {
+		log_error("trying to read when not enabled");
+		return -EIO;
+	}
+
+	if (req->msg.o.data != NULL && req->msg.o.size != READ_SIZE) {
+		return -EIO;
+	}
+
+	/* NOTE: mutex not needed for data races, but ensures no reader would be added while servicing single read */
+	mutexLock(ad7779_common.async.lock);
+
+	/* schedule async read */
+	req->state = stPendingRead;
+	*respond = 0;
+
+	condSignal(ad7779_common.async.cond);
+	mutexUnlock(ad7779_common.async.lock);
+
+	return EOK;
+}
+#else
```

Comment:
```suggestion


#else


```

Reply:
this looked strange, moved `ifdef` inside the function